README.md

```diff
@@ -142,15 +142,16 @@ Note that a widget instance may only be registered once. A factory is not allowe
 
 #### Seeing if an action, store or widget is registered
 
-To see whether a particular action, store or widget is registered, use the `has*()` methods:
+To see whether a particular action, store is registered, use the `has*()` methods:
 
 ```ts
 app.hasAction('my-action');
 app.hasStore('my-store');
 app.hasWidget('my-widget');
 ```
+The `hasAction` and `hasStore` methods return `true` if the respective item was registered, and `false` if not.
 
-Each method returns `true` if the respective item was registered, and `false` if not.
+The `hasWidget()` method returns a promise which will resolve to the `true` or `false` state, this is because not only does the `hasWidget()` method check the `instance` and `factory` registries it checks if the `getWidget()` method was called whether it would dynmically create the widget based on state associated to the `id`, see [Loading an action, store or widget](#Loading-an-action-store-or-widget).
```

Comment:
I would simplify this a little:

> The `hasWidget` method returns a promise, rather than a boolean. It is resolved with `true` if a widget instance or factory has been registered with that ID, or if a widget can be created based on the default widget store and registered custom elements. Otherwise it's resolved with `false`. See [Loading an action, store or widget](#loading-an-action-store-or-widget) for more.

(Also note that GitHub generates link anchors in lower case, so `loading-an-action-store-or-widget` rather than `Loading-an-action-store-or-widget`.